src/vs/workbench/contrib/files/browser/files.contribution.ts

```diff
@@ -362,6 +362,11 @@ configurationRegistry.registerConfiguration({
 			'description': nls.localize('fileDialogDefaultPath', "Default path for file dialogs, overriding user's home path. Only used in the absence of a context-specific path, such as most recently opened file or folder."),
 			'scope': ConfigurationScope.MACHINE
 		},
+		'files.dialog.useParentPathForOpenFolder': {
```

Comment:
I am still having some trouble with this setting. Because of this:
- it only applies to when there is a most recently used resource and not for the default path
- it only applies to folders, not files
- it only applies when opening folders, not files

Lets try to get some ideas. Would it be simpler if we did the inverse setting that is `false` by default?